ibm/service/kms/resource_ibm_kms_key_policies.go

```diff
@@ -289,6 +288,44 @@ func resourceIBMKmsKeyPolicyDelete(context context.Context, d *schema.ResourceDa
 
 }
 
+func resourceUpdatePolicies(context context.Context, d *schema.ResourceData, kpAPI *kp.Client, meta interface{}, key_id string) error {
+	var dualAuthEnable, rotationEnable bool
+	var rotationInterval int
+
+	policy := getPolicyFromSchema(d)
+
+	if policy.Rotation != nil {
+		rotationInterval = policy.Rotation.Interval
+		rotationEnable = *policy.Rotation.Enabled
+		if rotationInterval == 0 {
```

Comment:
can you add a comment at this point as to why we are checking this? something like `While updating a rotation policy, interval_month will be zero only when the user is not set this value and intend to only enable or disbale a policy. So, respective SDK functions are called here. In case, the user intends to update both, invoke function that updates both`

Reply:
Added comment